br/pkg/lightning/restore/check_info.go

```diff
@@ -139,6 +177,108 @@ func (rc *Controller) ClusterIsAvailable(ctx context.Context) error {
 	return nil
 }
 
+func (rc *Controller) checkEmptyRegion(ctx context.Context) error {
+	passed := true
+	message := "Cluster doesn't have too many empty regions"
+	defer func() {
+		rc.checkTemplate.Collect(Critical, passed, message)
+	}()
+
+	var result api.RegionsInfo
+	if err := rc.tls.WithHost(rc.cfg.TiDB.PdAddr).GetJSON(ctx, pdEmptyRegions, &result); err != nil {
+		return errors.Trace(err)
+	}
+	regions := make(map[uint64]int)
+	for _, region := range result.Regions {
+		for _, peer := range region.Peers {
+			regions[peer.StoreId]++
+		}
+	}
+
+	var messages []string
+	for storeID, regionCnt := range regions {
+		if regionCnt > errorEmptyRegionCntPerStore {
+			passed = false
+			messages = append(messages, fmt.Sprintf("TiKV store %v contains too many empty regions, "+
+				"the count must not exceed %v, but actual is %v", storeID, errorEmptyRegionCntPerStore, regionCnt))
+		} else if regionCnt > warnEmptyRegionCntPerStore {
+			messages = append(messages, fmt.Sprintf("TiKV store %v contains too many empty regions, "+
+				"the count should not exceed %v, but actual is %v", storeID, warnEmptyRegionCntPerStore, regionCnt))
+		}
+	}
+	if len(messages) > 0 {
+		message = strings.Join(messages, "\n")
+	}
+	return nil
+}
+
+// checkRegionDistribution checks if regions distribution is unbalanced.
+func (rc *Controller) checkRegionDistribution(ctx context.Context) error {
+	passed := true
+	message := "Cluster region distribution is balanced"
+	defer func() {
+		rc.checkTemplate.Collect(Critical, passed, message)
+	}()
+
+	result := &api.StoresInfo{}
+	err := rc.tls.WithHost(rc.cfg.TiDB.PdAddr).GetJSON(ctx, pdStores, result)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(result.Stores) <= 1 {
+		return nil
+	}
+	sort.Slice(result.Stores, func(i, j int) bool {
+		return result.Stores[i].Status.RegionCount < result.Stores[j].Status.RegionCount
+	})
+	minStore := result.Stores[0]
+	maxStore := result.Stores[len(result.Stores)-1]
+	if maxStore.Status.RegionCount <= checkRegionCntRatioThreshold {
+		return nil
+	}
+	if minStore.Status.RegionCount == 0 {
```

Comment:
What if the cluster is big but totally empty, can tidb ensure that every store contains some region?🤔️

Reply:
If the cluster is very empty, each store should contains very few regions (<= 1000). In this situation, check will be skipped before.

Reply:
so. If we change the ratio calculation to minRegionCount / maxRegionCount, we can avoid this extra check🤔️

Reply:
Makes sense!